### PR TITLE
Remove core extensions

### PR DESCRIPTION
**Pull request by matthewhanson**
**Related Issue(s):** #
#1063

**Proposed Changes:**

1. Remove core extensions (they are now in stac-extensions org)

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [ ] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

File: extensions/README.md [hunk 1]
```diff
@@ -41,44 +42,26 @@ the Item Asset objects contained in the Item, but may also be used in an individ
 
 ## Core STAC Extensions
 
-These extensions are considered stable, and are included directly in this repository.
+These extensions are considered stable and are widely used in many production implementations. As additional extensions advance
+through the [Extension Maturity](#extension-maturity) classification they will be added here.
```
Comment (philvarner) [inline extensions/README.md:46]:
```suggestion
through the [Extension Maturity](#extension-maturity) classification they, will be added here.
```

File: item-spec/item-spec.md [hunk 2]
```diff
@@ -147,7 +147,7 @@ to enable cross-catalog search on known fields.
 * [STAC Common Metadata](common-metadata.md#stac-common-metadata) - A list of fields commonly used
 throughout all domains. These optional fields are included for STAC Items by default.
 * [Extensions](../extensions/README.md#core-stac-extensions) - Additional fields that are more specific,
-such as [EO](../extensions/eo), [View](../extensions/view) and more.
+such as [EO](https://github.com/stac-extensions//eo), [View](https://github.com/stac-extensions/view) and more.
```
Comment (philvarner) [inline item-spec/item-spec.md:150]:
```suggestion
such as [EO](https://github.com/stac-extensions//eo), [View](https://github.com/stac-extensions/view).
```

**Review by cholmes — APPROVED** [review]

**Review by philvarner — APPROVED** [review]

File: item-spec/item-spec.md [hunk 3]
```diff
@@ -147,7 +147,7 @@ to enable cross-catalog search on known fields.
 * [STAC Common Metadata](common-metadata.md#stac-common-metadata) - A list of fields commonly used
 throughout all domains. These optional fields are included for STAC Items by default.
 * [Extensions](../extensions/README.md#core-stac-extensions) - Additional fields that are more specific,
-such as [EO](../extensions/eo), [View](../extensions/view) and more.
+such as [EO](https://github.com/stac-extensions//eo), [View](https://github.com/stac-extensions/view).
```
Comment (philvarner) [inline item-spec/item-spec.md:150]:
```suggestion
such as [EO](https://github.com/stac-extensions/eo), [View](https://github.com/stac-extensions/view).
```

File: best-practices.md [hunk 4]
```diff
@@ -338,9 +338,9 @@ actual role requirements.
...
-| local-incidence-angle, noise-power, amplitude, magnitude, sigma0, beta0, gamma0, date-offset, covmat, prd | [SAR Extension](https://github.com/stac-extensions/sar/README.md#best-practices) | See the [table](https://github.com/stac-extensions/sar/README.md#best-practices) in SAR for more information. , and the definitive list of roles related to SAR. |
+| reflectance, temperature, saturation, cloud, cloud-shadow | [EO Extension](https://github.com/stac-extensions/eo/blob/main/README.md#best-practices) | See the [table](https://github.com/stac-extensions/eo/blob/main/README.md#best-practices) in EO for more information, and the definitive list of roles related to EO. |
+| incidence-angle, azimuth, sun-azimuth, sun-elevation, terrain-shadow, terrain-occlusion, terrain-illumination | [View Extension](https://github.com/stac-extensions/view/blob/main/README.md#best-practices) | See the [table](https://github.com/stac-extensions/view/README.md#best-practices) in View for more information, and the definitive list of roles related to viewing angles. |
```
Comment (philvarner) [inline best-practices.md:342]:
```suggestion
| incidence-angle, azimuth, sun-azimuth, sun-elevation, terrain-shadow, terrain-occlusion, terrain-illumination | [View Extension](https://github.com/stac-extensions/view/blob/main/README.md#best-practices) | See the [table](https://github.com/stac-extensions/view/blob/main/README.md#best-practices) in View for more information, and the definitive list of roles related to viewing angles. |
```

**Review by cholmes — APPROVED** [review]

File: catalog-spec/catalog-spec.md [hunk 5]
```diff
@@ -1,7 +1,8 @@
-# STAC Catalog Specification
+# STAC Catalog Specification <!-- omit in toc -->
```
Comment (m-mohr) [inline catalog-spec/catalog-spec.md:1]: What is that?

Comment (m-mohr) [reply catalog-spec/catalog-spec.md:1]: Which ones do you use? Sounds useful.

File: catalog-spec/catalog-spec.md [hunk 6]
```diff
@@ -119,7 +119,7 @@ The following table lists the Media Types to use for STAC structures.
 | Media Type             | Description                                                |
 | ---------------------- | ---------------------------------------------------------- |
 | `application/geo+json` | A STAC [Item](../item-spec/item-spec.md)                   |
-| `application/json`     | A STAC [Catalog](#stac-catalog-specification)              |
+| `application/json`     | A STAC [Catalog](#)              |
```
Comment (m-mohr) [inline catalog-spec/catalog-spec.md:122]: Why not just remove the link?

Comment (philvarner) [reply catalog-spec/catalog-spec.md:122]: I didn't think too much about the usefulness -- I was just trying to fix the validation error. 
```suggestion
| `application/json`     | A STAC Catalog            |
```

File: catalog-spec/catalog-spec.md (hunk 5 above)
Comment (philvarner) [reply catalog-spec/catalog-spec.md:1]: It's Visual Studio Code "Markdown All in One" plugin. I have on my list to add it to CONTRIBUTING.md 

I had to add a TOC to Item, so I added `- [Overview](#overview)`, and then the rest of it was populated when I saved by the plugin.

**Pull request merged**